guvnor-m2repo-editor/guvnor-m2repo-editor-backend/src/main/java/org/guvnor/m2repo/backend/server/GuvnorM2Repository.java

```diff
@@ -509,11 +513,9 @@ private RemoteRepository getRemoteRepoFromDeployment( DeploymentRepository repo
         Server server = settings.getServer( repo.getId() );
 
         if ( server != null ) {
-            remoteRepoBuilder.setAuthentication(
-                    new AuthenticationBuilder()
-                            .addUsername( server.getUsername() )
-                            .addPassword( server.getPassword() )
-                            .build() );
+            Authentication authentication = embedder.getMavenSession().getRepositorySession()
+                    .getAuthenticationSelector().getAuthentication( remoteRepoBuilder.build() );
```

Comment:
Drools PR has been merged. I'll look at this tomorrow (Thursday GMT).
Thanks for the PR.. it'll be great to cover your use case.